src/NServiceBus.Core/Diagnostics/ActivityDecorator.cs

```diff
@@ -0,0 +1,60 @@
+namespace NServiceBus
+{
+    using System.Collections.Generic;
+    using System.Diagnostics;
+
+    class ActivityDecorator
+    {
+        public static void PromoteHeadersToTags(Activity activity, Dictionary<string, string> headers)
+        {
+            if (activity == null)
+            {
+                return;
+            }
+
+            foreach (var header in headers)
+            {
+                if (HeaderMapping.TryGetValue(header.Key, out var tagName))
+                {
+                    activity.AddTag(tagName, header.Value);
+                }
+            }
+        }
+
+        internal static readonly Dictionary<string, string> HeaderMapping = new()
+        {
+            { Headers.SagaId, "nservicebus.saga_id" },
+            { Headers.MessageId, "nservicebus.message_id" },
+            { Headers.CorrelationId, "nservicebus.correlation_id" },
+            { Headers.ReplyToAddress, "nservicebus.reply_to_address" },
+            { Headers.NServiceBusVersion, "nservicebus.version" },
+            { Headers.ControlMessageHeader, "nservicebus.control_message" },
+            { Headers.SagaType, "nservicebus.saga_type" },
```

Comment:
`SagaType`, `IsSagaTimeoutMessage` and `SagaId` headers could probably be namespaced into `nservicebus.saga.[...]`?